src/visualizers/panels/ExecutionIndex/ExecutionIndexControl.js

```diff
@@ -38,45 +41,88 @@ define([
     };
 
     ExecutionIndexControl.prototype._initWidgetEventHandlers = function () {
-        this._widget.setExecutionDisplayed = this.setExecutionDisplayed.bind(this);
+        this._widget.setDisplayedExecutions = this.setDisplayedExecutions.bind(this);
     };
 
-    ExecutionIndexControl.prototype.setExecutionDisplayed = function (id, bool) {
-        var lines = this._linesForExecution[id] || [],
-            otherLines,
-            wasMultiLine = this.displayedExecCount() > 1,
-            isMultiLine;
+    ExecutionIndexControl.prototype.setDisplayedExecutions = function (beforeClickIds, afterClickIds) {
```

Comment:
This is a little odd to me. It seems that `setExecutionDisplayed` (which accepted the updated execution as an argument) has been updated to accept the before and after states. However, this method then uses the states to determine the execution which has changed (which was the old argument).

If you want to determine the execution id that changed, why aren't you just keeping the old method signature rather than converting it to a method which accepts states and recomputes it?